hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/util/TokenRelListCalculator.java

```diff
@@ -111,29 +111,31 @@ public TokenRelsRemovalResult removeTokenRels(
             @NonNull final Account account, @NonNull final List<TokenRelation> tokenRelsToDelete) {
         // Precondition: verify all token relation objects have the same account number as the given account object
         if (tokenRelsToDelete.stream()
-                .anyMatch(tokenRel -> tokenRel != null && tokenRel.accountNumber() != account.accountNumber())) {
+                .anyMatch(tokenRel ->
+                        tokenRel != null && !tokenRel.accountId().equals(asAccount(account.accountNumber())))) {
```

Comment:
Might be cleaner as follows:
`!tokenRel.accountId().accountNumber() == account.accountNumber())) {`